### ISSUE TITLE
Detect incorrect injection of libraries into chef classes

### ISSUE THREAD

**Issue by tas50** [chef/cookstyle#274]
### What category of cop is this?:

Please try to select as few as possible.

- [ ] Deprecation - A cop that alerts to a deprecation in the Chef Infra Client
- [ ] Correctness - A cop that alerts when incorrect coding behavior is being used
- [ ] Sharing - A cop that alerts to missing functionality necessary for sharing cookboooks with others
- [ ] Style - A cop that alerts to a style best practice
- [ ] Simplification - A cop that alerts to ways code can be simplified through refactoring
- [ ] Effortless - A cop that alerts on code that must be resolved to move to the Effortless pattern
- [ ] Policyfiles - A cop that alerts on code that must be resolved to move to the Policyfiles pattern

### Describe the new cop:
There are many ways people inject library code into their resources. Most of the time it's wrong and has been cargo culted over the years. We should provide guidance on doing the right thing.

### What it would trigger on:

```ruby
 ::Chef::Recipe.send(:include, Filebeat::Helpers)
 ::Chef::Resource.send(:include, Filebeat::Helpers)
::Chef::Provider.send(:include, Filebeat::Helpers)
```

### What it would autocorrect to if applicable

```ruby
 ::Chef::Recipe.send(:include, Filebeat::Helpers)

OR

 ::Chef::Providers.send(:include, Filebeat::Helpers)
```

becomes

```ruby
 ::Chef::DSL::Recipe.send(:include, Filebeat::Helpers)
```

AND

```ruby
::Chef::Resource.send(:include, Filebeat::Helpers)
```
becomes

```ruby
::Chef::DSL::Resource.send(:include, Filebeat::Helpers)
```

**Comment by tas50** [comment]
We should also move to the new format of injecting the include: https://github.com/rubocop-hq/rubocop/issues/7274

**Comment by tas50** [comment]
Closed by #338
